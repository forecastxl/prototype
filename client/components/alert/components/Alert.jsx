import React, { PropTypes } from 'react'
import AlertWrapper from './AlertWrapper'

function Alert({ messages }) {
  return (
    <AlertWrapper>
      {messages.map(message => <p key={message}>{message}</p>)}
    </AlertWrapper>
  )
}

Alert.propTypes = {
  messages: PropTypes.array.isRequired
}

export default Alert
