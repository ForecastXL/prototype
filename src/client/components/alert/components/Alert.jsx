import React from 'react'
import { string, oneOf } from 'prop-types'
import AlertWrapper from './AlertWrapper'
import AlertIcon from './AlertIcon'

function Alert({ message, type }) {
  return (
    <AlertWrapper type={type}>
      <AlertIcon type={type} />
      <span>{message}</span>
    </AlertWrapper>
  )
}

Alert.propTypes = {
  message: string.isRequired,
  type: oneOf(['error', 'warning', 'info', 'success']).isRequired
}

export default Alert
