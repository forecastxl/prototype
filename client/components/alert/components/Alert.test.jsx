import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Alert from './Alert'

describe('<Alert />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Alert
        messages={['message', 'message']}
      />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
