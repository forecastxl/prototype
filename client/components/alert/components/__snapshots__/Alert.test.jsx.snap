exports[`<Alert /> renders correctly 1`] = `
<MuiComponent>
  <p>
    message
  </p>
  <p>
    message
  </p>
</MuiComponent>
`;
